src/server.js

```diff
@@ -258,7 +258,11 @@ function initializeClient(socket, client, token, lastMessage) {
 	socket.on(
 		"more",
 		function(data) {
-			client.more(data);
+			const history = client.more(data);
+
+			if (history !== null) {
```

Comment:
You could have left `return;` above and here do `if (history)` or `if (history !== undefined)`, but that's pretty much the same thing and I get the preference for being explicit.